controller/jobs.go

```diff
@@ -237,6 +237,13 @@ func (c *controllerAPI) PutJob(ctx context.Context, w http.ResponseWriter, req *
 	}
 
 	job.AppID = app.ID
+
+	schema := schemaForType(ct.Job{})
```

Comment:
What do you think about a `schema.Validate(thing interface{})` that would join the two calls?

```go
func Validate(thing interface{}) err {
    schema := schemaForType(thing)
    if err := validateInput(schema, thing); err != nil {
        return
    }
}
```